ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/messagelistpage/helpers/OffboardingHelperTest.java

```diff
@@ -0,0 +1,335 @@
+        underTest.getOffboardingListItems(null, true, offboardingHelperViewCallback);
+    }
+
+    @Test
+    public void givenGetOffboardingListItemsWhenConversationNotCompletedThenEmptyList() {
+        // Act
+        List<BaseListItem> result = underTest.getOffboardingListItems("name", false,
+                offboardingHelperViewCallback);
+        // Assert
+        assertEquals(Collections.EMPTY_LIST, result);
+    }
+
+    @Test
+    public void
+    givenGetOffboardingListItemsWhenCurrentRatingSubmittedViaUINullThenGetOffboardingItemsToSelectRating() {
+        // Arrange
+        Rating.SCORE currentRatingSubmittedViaUI = null;
+        setInternalState(underTest, "currentRatingSubmittedViaUI", currentRatingSubmittedViaUI);
+        when(Kayako.getApplicationContext()).thenReturn(context);
+        when(context.getString(R.string.ko__messenger_input_feedback_rating_instruction_message))
+                .thenReturn("message");
+        // Act
+        List<BaseListItem> result = underTest.getOffboardingListItems("name", true,
+                offboardingHelperViewCallback);
+        // Assert
+        assertTrue(result.get(0) instanceof InputFeedbackRatingListItem);
+    }
+
+    @Test
+    public void
+    givenGetOffboardingListItemsWhenCurrentFeedSubmittedViaUINullThenGetOffboardingItemsToSelectRating() {
+        // Arrange
+        Rating.SCORE currentRatingSubmittedViaUI = Rating.SCORE.GOOD;
+        setInternalState(underTest, "currentRatingSubmittedViaUI", currentRatingSubmittedViaUI);
+        String currentFeedbackSubmittedViaUI = null;
+        setInternalState(underTest, "currentFeedbackSubmittedViaUI", currentFeedbackSubmittedViaUI);
+        when(Kayako.getApplicationContext()).thenReturn(context);
+        when(context.getResources()).thenReturn(resources);
+        when(resources.getString(R.string
+                .ko__messenger_input_feedback_rating_instruction_message)).thenReturn("message");
+        // Act
+        List<BaseListItem> result = underTest.getOffboardingListItems("name", true,
+                offboardingHelperViewCallback);
+        // Assert
+        assertFalse(result.isEmpty());
+    }
+
+    @Test
+    public void
+    givenGetOffboardingListItemsWhenFeedbackAndRatingNotNullThenGetOffboardingItemsToSelectRating
+            () {
+        // Arrange
+        Rating.SCORE currentRatingSubmittedViaUI = Rating.SCORE.GOOD;
+        setInternalState(underTest, "currentRatingSubmittedViaUI", currentRatingSubmittedViaUI);
+        String currentFeedbackSubmittedViaUI = "good";
+        setInternalState(underTest, "currentFeedbackSubmittedViaUI", currentFeedbackSubmittedViaUI);
+        // Act
+        List<BaseListItem> result = underTest.getOffboardingListItems("name", true,
+                offboardingHelperViewCallback);
+        // Assert
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    public void givenResetRatingAddedByUserThenIdOfLastRatingSetToZero() throws Exception {
```

Comment:
✅ Good to see that test methods respect Java lower-camel-case and that Exception is thrown instead of specific exceptions.

This way the code respects Java naming conventions for methods and will avoid the risk of becoming "brittle" because they must be changed whenever the method under test changes.

**No change suggested, looks good!**	  

<a id="e0497d70-5328-11e8-8cbd-e396d4991aea" href="none" codereview-uuid-reserved-tag />